File: packages/var-runtime/bundles/02-context-isolation/counter.steps.ts
import { defineState } from '@oselvar/var-runtime'

const { action, sensor } = defineState<{ count: number }>(() => ({ count: 0 }))

action('I increment', (ctx) => {
  ctx.count += 1
})

sensor('The count is {int}', (ctx, n: number) => {
  if (ctx.count !== n) throw new Error(`expected ${n} but got ${ctx.count}`)
})

File: conformance/bundles/02-context-isolation/counter.steps.rb
require "oselvar/var"

param, stimulus, sensor = steps { { count: 0 } }

stimulus.("I increment") { |state| { count: state[:count] + 1 } }

sensor.("The count is {int}") do |state, n|
  raise "expected #{n} but got #{state[:count]}" if state[:count] != n
end
